app/utils.py

```diff
@@ -0,0 +1,25 @@
+from datetime import datetime
+from functools import partial
+from calendar import timegm
+from time import strptime
+from settings import FORMATS
+
+
+def check_datetime_format(format, date: str):
+    try:
+        return bool(datetime.strptime(date, format))
+    except ValueError:
+        return False
+
+
+def convert_time_range(time_range: list):
+    try:
+        new_time_range = []
+        for time_ in time_range:
+            index = list(
+                map(partial(check_datetime_format, date=time_), FORMATS)
+            ).index(True)
+            new_time_range.append(timegm(strptime(time_, FORMATS[index])) * 1000)
```

Comment:
Why use `timegm` instead of `datetime.datetime.fromtimestamp()`?

Reply:
Because i convert to timestamp from normal date

Reply:
What is normal date?

Reply:
e.g 2019-10-18

Reply:
But the result of `strptime()` function is `datetime`

```
classmethod datetime.strptime(date_string, format)
Return a datetime corresponding to date_string, parsed according to format.
```

https://docs.python.org/3/library/datetime.html#datetime.datetime.fromisocalendar

Reply:
For fromtimestamp method you need to pass arguments in int type. For example fromtimestamp(2019, 10, 18) but variable in str type('2019-10-19'). According to this, before use fromtimestamp you need to parse variable

Reply:
Function `fromtimestamp()` is used to convert from `timestamp` into `datetime` object. It does not work like this `fromtimestamp(2019, 10, 18)` and accepts only single arguments, that is unix timestamp value (number of seconds from `1970-01-01` in specific timezone, by default UTC).
Contrary `datetime.timestamp()` can be used to convert from `datetime` into `timestamp`.

Reply:
Ok, i tested it. You are right about fromtimestamp. But datetime.timestamp() returns time in local time. There is no info about time in what format(local or GMU/UTC) it is. So i decided to use in GMU/UTC.

Reply:
https://docs.python.org/3/library/datetime.html#datetime.datetime.timestamp

```
timestamp = dt.replace(tzinfo=timezone.utc).timestamp()
```